staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -407,11 +416,34 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	case <-localError:
 	}
 
+	// reset dataStream to discard any unsent data, preventing port forwarding from being blocked.
+	// we must reset dataStream before waiting on errorChan, otherwise,
+	// the blocking data will affect errorStream and cause <-errorChan to block indefinitely.
+	_ = dataStream.Reset()
+
 	// always expect something on errorChan (it may be nil)
 	err = <-errorChan
 	if err != nil {
 		runtime.HandleError(err)
-		pf.streamConn.Close()
+		// don't tear down the whole parent port-forward pf.streamConn when there's
+		// an error handling a single request, container runtime should handle
+		// the streaming error by reseting the connection, so we should be able
+		// to keep the connection open until a user explicitly requests an end
+
+		// This also forces drain of the errorStream, similar to dataStream above
+		_ = errorStream.Reset()
```

Comment:
does resetting errorStream from the client actually accomplish anything? we already called `Close()` on our side of the errorStream (immediately after we opened it), we never send data from the client over errorStream so there's nothing to drain, and we already received from `errorChan` which means `io.ReadAll(errorStream)` completed so the server already closed their end as well, right?

Reply:
That makes sense, we only use this as read only, and can safely drop this reset.

Reply:
This was removed.